### PR TITLE
Feature/extract signup flow

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Move the new signup code to a separate library, [WordPress-Login-Flow-Android](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android).  This is the second part of the [login library extraction](https://github.com/wordpress-mobile/WordPress-Android/pull/6935).

### Test
These changes cover all aspects of the signup process except the epilogue.  To test fully, each signup method (i.e. email and social) and flow (i.e. automated email, manual email, magic link, manual password) should be tried.  Attempting to signup with an existing account moves from signup to login, but it should be included in testing as well.

### CODE REVIEW

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupEmailFragment.java [hunk 1]
```diff
@@ -264,7 +260,7 @@ public void onAvailabilityChecked(OnAvailabilityChecked event) {
                             if (event.isAvailable) {
                                 mLoginListener.showSignupMagicLink(event.value);
                             } else {
-                                AnalyticsTracker.track(AnalyticsTracker.Stat.SIGNUP_EMAIL_TO_LOGIN);
+                                mAnalyticsListener.trackAnalyticsSignIn(mAccountStore, mSiteStore, true);
```
Comment (aforcier) [inline libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupEmailFragment.java]: Shouldn't this be `trackSignupEmailToLogin()`?

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupGoogleFragment.java [hunk 2]
```diff
@@ -38,7 +36,7 @@
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        ((WordPress) getActivity().getApplication()).component().inject(this);
+        AndroidSupportInjection.inject(this);
```
Comment (aforcier) [inline libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupGoogleFragment.java]: This injector call should be done in `onAttach()` instead.

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupEmailFragment.java (hunk 1 above)
Comment (theck13) [reply libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupEmailFragment.java]: Yep.  I updated that statement in d20d82a.

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupGoogleFragment.java (hunk 2 above)
Comment (theck13) [reply libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupGoogleFragment.java]: I moved the injector call in d8894f1.

**Comment by aforcier** [comment]
Gave this a run through in the Woo app, looks not to be causing any issues. 👍 

I wonder if moving the [signup-related methods](https://github.com/wordpress-mobile/WordPress-Android/blob/dabb1410403f03b8dba66c7a13985a4616bfcc4e/libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginListener.java#L70-L74) out of `LoginListener` and over to a `SignupListener` might make sense, to allow the host application to ignore the listener instead of having to stub out those methods (if they're not implementing signup). But this is pretty minor, I'll leave it up to you.

For my end, will test this a bit WordPress-side tomorrow.

**Comment by theck13** [comment]
I teetered about separating login and signup throughout the library (e.g. things like `LoginFragmentModule` and `SignupFragmentModule`), but I ended up keeping everything together for simplicity.  However, something like a `SignupListener` might be helpful for the reason you mentioned.

**Comment by hypest** [comment]
Looks good to me @theck13 !

Tried:
1. ✅ Signup via magiclink email
2. ✅ Signup via google social

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupGoogleFragment.java [hunk 3]
```diff
@@ -35,10 +33,9 @@
 
     public static final String TAG = "signup_google_fragment_tag";
 
-    @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        ((WordPress) getActivity().getApplication()).component().inject(this);
+    @Override public void onAttach(Context context) {
+        super.onAttach(context);
```
Comment (aforcier) [inline libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupGoogleFragment.java]: `AndroidSupportInjection.inject(this);` needs to come above `super.onAttach()` (per the [Dagger docs](https://google.github.io/dagger/android.html)).

Comment (theck13) [reply libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupGoogleFragment.java]: I moved the statement in 883bafb.

**Comment by aforcier** [comment]
:shipit: from me! Over to you, @hypest.

**Comment by hypest** [comment]
It's a :shipit: from this side too! I will merge.

**Pull request merged**